### PR TITLE
Adjust regex to ensure a starts with on URL pattern protocol.

### PR DESCRIPTION
**Pull request by mathewjordan**
# Summary 
URL Pattern matching was checking for any part of the string input into the URL field to start with `http://` or `https://`. In that case an input value of `foo https://bar.baz` would validate TRUE. This now validates false and will require a url to match a pattern `https://foo.bar` with only valid starts:

- `http://foo.bar`
- `https://foo.bar`
- `www.foo.bar`

_My solution here probably could be done completely with RegEx but I'm not sure how considering we are seeking to allow **www** as well as the protocols to be valid._

# Specific Changes in this PR
- Adjusts URL validation to make sure URL strings start with either `http://`, `https://`, or `www`

# Version bump required by the PR

See [Semantic Versioning 2.0.0](https://semver.org/) for help discerning which is required.

- [x] Patch
- [ ] Minor
- [ ] Major

# Steps to Test

1. Edit any work
2. Add a Related URL of `foo https://bar.baz`
3. This should fail validation ❌ 
2. Add a Related URL of `https://foo.bar`
3. This should pass validation ✅ 

# :rocket: Deployment Notes

**Note** - __if you check any of these boxes go to the [(always open) `main` <- `staging` PR](https://github.com/nulib/meadow/pulls) and add detailed notes and instructions to help out others who may end up deploying your changes to production__

- Backward compatible API changes
  - [ ] Database Schema changes
  - [ ] GraphQL API
  - [ ] Elasticsearch API
  - [ ] Ingest Sheet
  - [ ] CSV metadata export/update API
  - [ ] Shared Links export API
- Backwards-incompatible API changes
  - [ ] Database Schema changes
  - [ ] GraphQL API
  - [ ] Elasticsearch API
  - [ ] Ingest Sheet
  - [ ] CSV metadata export/update API
  - [ ] Shared Links export API
- [ ] Requires data migration
- [ ] Requires database triggers disabled during deployment/migration
- [ ] Requires reindex
- [ ] Terraform changes
  - [ ] Adds/requires new or changed Terraform variables
- [ ] Pipeline configuration changes (requires `mix meadow.pipeline.setup` run)
- [ ] Requires new variable added to `miscellany`
- [ ] Specific deployment synchronization instructions with other apps/API's
- [ ] Other specific instructions/tasks

# Tested/Verified
- [ ] End users/stakeholders

### CODE REVIEW

**Comment by mathewjordan** [comment]
@adamjarling For your eyes when you return.

**Review by adamjarling — CHANGES_REQUESTED** [review]
@mathewjordan This looks good for Batch Edit, however can we apply the same error message display to a single Work Edit?   

To test: in Work Edit, enter "asdf http://google.com" for "Related Url".  The form doesn't submit (which is probably good), but guessing the validation error just isn't getting communicated in the UI?

File: assets/js/services/helpers.js [hunk 1]
```diff
@@ -24,7 +24,10 @@ export function formatSimpleISODate(date) {
 }
 
 export function isUrlValid(url) {
-  return url.match(URL_PATTERN_MATCH) ? true : false;
+  return url.match(URL_PATTERN_MATCH) &&
+    URL_PATTERN_START.some((validStart) => url.startsWith(validStart))
+    ? true
```
Comment (adamjarling) [inline assets/js/services/helpers.js:29]: You probably don't even need lines 29-30 here.  Guessing it'll coerce to truthy?

**Review by mathewjordan — COMMENTED** [review]
@adamjarling I dug into this a bit more to try and remember what we are doing here. It looks like all the validation here happens during the attempted submission process. Do we want to switch this up so that some minimal feedback is given during each onChange?

**Comment by adamjarling** [comment]
@mathewjordan Either way as long as it's good UX, I'm fine pre-handling the submission or the alternative was React Hook Form catches the validation error and displays the error message prior to sending off the network request?

**Pull request closed without merging**